source/common/router/router.h

```diff
@@ -288,6 +288,7 @@ class Filter : Logger::Loggable<Logger::Id::router>,
     void decodeHeaders(Http::HeaderMapPtr&& headers, bool end_stream) override;
     void decodeData(Buffer::Instance& data, bool end_stream) override;
     void decodeTrailers(Http::HeaderMapPtr&& trailers) override;
+    void decodeMetadata(Http::MetadataMap&) override { NOT_REACHED_GCOVR_EXCL_LINE; }
```

Comment:
I think we're going to need to implement something here.  Maybe a TODO instead of marking it as not reached?

Reply:
Done.